.travis.yml

```diff
@@ -57,51 +67,68 @@ matrix:
     - &mac
       os: osx
       osx_image: xcode9.2
-      if: branch != master
+      if: type != pull_request
       env: MACOSX_DEPLOYMENT_TARGET=10.7 TARGET=x86_64-apple-darwin NO_ADD=1
     - <<: *mac
+      if: branch != master
       env: MACOSX_DEPLOYMENT_TARGET=10.7 TARGET=i686-apple-darwin
 
     - name: rustup-init.sh on CentOS 6
       language: minimal
       install:
       script:
         - docker run
-            --volume "$TRAVIS_BUILD_DIR":/checkout:ro
-            --workdir /checkout
-            --rm
-            -it
-            centos:6
-            sh ./ci/raw_init.sh
+          --volume "$TRAVIS_BUILD_DIR":/checkout:ro
+          --workdir /checkout
+          --rm
+          -it
+          centos:6
+          sh ./ci/raw_init.sh
       before_deploy:
       deploy:
 
+    - &windows
+      os: windows
+      env: TARGET=x86_64-pc-windows-msvc NO_ADD=1 EXE_EXT=.exe
+      services: nil
+      language: bash
+      before_install:
+        - choco install strawberryperl
+        - export PATH="/c/Strawberry/perl/bin:$PATH"
```

Comment:
Did you install `strawberryperl` because of failing to build openssl?
On Windows we don't need to build with openssl feature. You could 
detect OS in `ci/run.sh`  and run appropriate build command .

Reply:
Use `TRAVIS_OS_NAME` env: https://docs.travis-ci.com/user/environment-variables#default-environment-variables

Reply:
We install strawberryperl because openssl can't configure with the git-bash perl, it's too broken - why would we not build with openssl on windows?

Reply:
Because Windows users want to use Windows builtin TLS library. Also
in Appveyor build file we don't use OpenSSL feature either.

Reply:
Oh?  I must have missed that - what feature do I pass to enable that, because if I pass no feature flags, it still tries to build openssl.

Reply:
The line 11 use openssl feature:
https://github.com/rust-lang/rustup.rs/blob/04039b0d13a8f5fba70dd55a9dcca0a52a41f0ad/ci/run.sh#L8-L17

You could use `TRAVIS_OS_NAME` env variable and get rid of that feature 
on Windows.

Reply:
Mmm, when I was trying to get it to work without strawberryperl I tried removing the features there but it still seemed to be building openssl anyway.  I'll try again I guess.